docs/adr/adr-007-data-retention.md

```diff
@@ -0,0 +1,196 @@
+example of this can be seen if we look at subscriber lists. We have over
+~73,000 lists with active subscribers but in the last week alone, we've
+accumulated 1139 lists which don't have any subscriptions at all (ended or
+active). These unused lists prevent us from doing simple auditing tasks
+easily, like finding which lists to report on.
+
+Knowing the above, we thought about how we could cleanup this unused data to
+keep EmailAlertApi's database manageable going forward. There are a couple of
+different approaches to consider, including storing unused data in cloud
+storage away from the system's database (like we do when archiving emails) and
+subsequently deleting the data locally. Or, setting up a means of removing the
+data we don't use from the system entirely.
+
+[archived]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/email_archive_worker.rb
+[deleted]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/email_deletion_worker.rb
+[initial policy]: https://github.com/alphagov/email-alert-api/blob/master/docs/adr/adr-003-data-retention.md
+[migrate subscribers to daily digest emails]: https://github.com/alphagov/email-alert-api/pull/1352
+
+## Decision
+
+We decided that historic, unused data should be removed periodically from the
+system entirely. It was concluded that the data the system no longer uses
+should only exist for a maximum of 1 year before removal. We sometimes
+analyse historic data from up to a year ago but we have no evidence of needing
+this data for analytics beyond a year.
+
+Emails (and their associated SubscriptionContents) are the one place where we
+already manage retention - these are held locally for 7 days, and also
+archived to Athena indefinitely. We still need this low retention for Emails in
+order to manage the size of the table. However, the archiving to Athena is
+inconsistent with the above decision to delete unused data entirely. While we
+originally thought Athena might serve as a data source for analytics, this has
+not happened in practice, so the archive data stored there is entirely unused.
+We therefore decided to remove the archiving mechanism for Emails, since it has
+no value.
+
+The following list documents our approach to the decision for different data
+in the system.
+
+* **ContentChange, Message & DigestRun**. Remove once their `created_at` value is
+  over a year old.
+
+* **MatchedContentChange, MatchedMessage & DigestRunSubscriber**. Cascaded when
+  their parent content change / message / digest run is deleted.
+
+* **Subscription**. Remove subscriptions when they've been ended for over a
+  year, based on their `ended_at` value. Once a subscription is ended it [can't
+  currently be made active again] and so can be deleted.
+
+* **Previously subscribed to SubscriberList**. Remove lists when they have no
+  active subscriptions and their latest, ended subscription is over a year old.
```

Comment:
```suggestion
  subscriptions (after removing historic ones) and they are over a year old.
```

Minor: this is different to [the actual implementation](https://github.com/alphagov/email-alert-api/blob/master/app/workers/historical_data_deletion_worker.rb#L56).

Reply:
The implementation you've linked to is concerning historic subscribers, rather then historic subscriber lists so I've the wording as is. 👍

Reply:
Oh sorry. I meant [here](https://github.com/alphagov/email-alert-api/blob/master/app/workers/historical_data_deletion_worker.rb#L46) - it is different.

Reply:
The ordering of the deletions was factored into the implementation to simply the code.

We decided that we should only remove subscriber lists which have no active subscriptions and the newest subscription is over a year old. As we delete the historic ended subscriptions prior to subscriber lists we don't have to include this extra logic in the code.

I could explain the way it was implemented but that seems to detract from the decision that was made. Does that make sense?

Reply:
Yes, that's fair. It is just a minor point, and I wanted to make sure you were aware of the difference. I agree it's a valid perspective on the policy.